featuretools/entityset/entity.py

```diff
@@ -284,40 +281,42 @@ def delete_variables(self, variable_ids):
             self.variables.remove(v)
 
     def set_time_index(self, variable_id, already_sorted=False):
-        # check time type
-        if not isinstance(self.df, pd.DataFrame) or self.df.empty:
-            time_to_check = vtypes.DEFAULT_DTYPE_VALUES[self[variable_id]._default_pandas_dtype]
-        else:
-            time_to_check = self.df[variable_id].iloc[0]
-        time_type = _check_time_type(time_to_check)
-
-        if time_type is None:
-            raise TypeError("%s time index not recognized as numeric or"
-                            " datetime" % (self.id))
+        self.time_index = variable_id
+        self._already_sorted = already_sorted
+        self._check_time_index()
 
-        if self.entityset.time_type is None:
-            self.entityset.time_type = time_type
-        elif self.entityset.time_type != time_type:
-            raise TypeError("%s time index is %s type which differs from"
-                            " other entityset time indexes" %
-                            (self.id, time_type))
+    def _check_time_index(self):
+        if self.time_index is None:
+            return
 
+        time_type = self._get_time_type()
         if is_instance(self.df, (dd, ks), 'DataFrame'):
             t = time_type  # skip checking values
-            already_sorted = True  # skip sorting
+            self._already_sorted = True  # skip sorting
         else:
             t = vtypes.NumericTimeIndex
-            if col_is_datetime(self.df[variable_id]):
+            if col_is_datetime(self.df[self.time_index]):
                 t = vtypes.DatetimeTimeIndex
 
         # use stable sort
-        if not already_sorted:
+        if not self._already_sorted:
             # sort by time variable, then by index
-            self.df = self.df.sort_values([variable_id, self.index])
+            self.df = self.df.sort_values([self.time_index, self.index])
+        self.convert_variable_type(self.time_index, t, convert_data=False)
 
-        self.convert_variable_type(variable_id, t, convert_data=False)
+    def _get_time_type(self, variable_id=None):
```

Comment:
We should move this outside of Entity, since we'll still want this after Entity has been replaced

Reply:
✅  d2b0c7c4